File: Backend_node/src/utils/ApiError.ts
class ApiError extends Error {
    public statusCode: number;
    public success: boolean;

    constructor(status: number, message = "something went wrong") {
        super(message);
        this.statusCode = status;
        this.message = message;
        this.success = false;
        
        // Ensure the prototype is correctly set for built-in Error extension
        Object.setPrototypeOf(this, ApiError.prototype);
    }
}

export default ApiError;

File: server/api/utils/api_error.py
class ApiError(Exception):
    def __init__(
        self,
        status_code: int,
        message: str = "Something went wrong",
        errors: list = [],
        stack: str = ""
    ):
        super().__init__(message)
        self.status_code = status_code
        self.message = message
        self.errors = errors
        self.data = None
        self.success = False

        if stack:
            self.stack = stack
